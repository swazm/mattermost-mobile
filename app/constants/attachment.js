// Copyright (c) 2015-present SWAZM, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const MAX_ATTACHMENT_FOOTER_LENGTH = 300;
