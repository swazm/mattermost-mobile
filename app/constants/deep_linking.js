// Copyright (c) 2015-present SWAZM, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    CHANNEL: 'channel',
    PERMALINK: 'permalink',
    OTHER: 'other',
};
