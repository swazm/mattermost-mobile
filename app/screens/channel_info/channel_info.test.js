// Copyright (c) 2015-present SWAZM, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';
import {General} from 'mattermost-redux/constants';

import ChannelInfo from './channel_info';

// ChannelInfoRow expects to receive the pinIcon as a number
jest.mock('assets/images/channel_info/pin.png', () => {
    return 1;
});

jest.mock('app/utils/theme', () => {
    const original = require.requireActual('app/utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('channel_info', () => {
    const intlMock = {
        formatMessage: jest.fn(),
        formatDate: jest.fn(),
        formatTime: jest.fn(),
        formatRelative: jest.fn(),
        formatNumber: jest.fn(),
        formatPlural: jest.fn(),
        formatHTMLMessage: jest.fn(),
        now: jest.fn(),
    };
    const baseProps = {
        canDeleteChannel: true,
        canConvertChannel: true,
        canManageUsers: true,
        viewArchivedChannels: true,
        canEditChannel: true,
        currentChannel: {
            id: '1234',
            display_name: 'Channel Name',
            type: General.OPEN_CHANNEL,
            create_at: 123,
            delete_at: 0,
            header: '',
            purpose: 'Purpose',
            group_constrained: false,
        },
        currentChannelCreatorName: 'Creator',
        currentChannelMemberCount: 2,
        currentChannelGuestCount: 0,
        currentUserId: '1234',
        currentUserIsGuest: false,
        isChannelMuted: false,
        ignoreChannelMentions: false,
        isCurrent: true,
        isFavorite: false,
        status: 'status',
        theme: Preferences.THEMES.default,
        isBot: false,
        isLandscape: false,
        actions: {
            clearPinnedPosts: jest.fn(),
            closeDMChannel: jest.fn(),
            closeGMChannel: jest.fn(),
            convertChannelToPrivate: jest.fn(),
            deleteChannel: jest.fn(),
            getChannelStats: jest.fn(),
            getChannel: jest.fn(),
            leaveChannel: jest.fn(),
            loadChannelsByTeamName: jest.fn(),
            favoriteChannel: jest.fn(),
            unfavoriteChannel: jest.fn(),
            getCustomEmojisInText: jest.fn(),
            selectFocusedPostId: jest.fn(),
            updateChannelNotifyProps: jest.fn(),
            selectPenultimateChannel: jest.fn(),
            setChannelDisplayName: jest.fn(),
            handleSelectChannel: jest.fn(),
        },
    };

    test('should match snapshot', async () => {
        const wrapper = shallow(
            <ChannelInfo
                {...baseProps}
            />,
            {context: {intl: intlMock}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should render convert to private button when user has team admin permissions', async () => {
        const wrapper = shallow(
            <ChannelInfo
                {...baseProps}
            />,
            {context: {intl: intlMock}},
        );

        const instance = wrapper.instance();
        const render = instance.renderConvertToPrivateRow();
        expect(render).toBeTruthy();
    });

    test('should not render convert to private button when user is not team admin or above', async () => {
        const wrapper = shallow(
            <ChannelInfo
                {...baseProps}
                canConvertChannel={false}
            />,
            {context: {intl: intlMock}},
        );

        const instance = wrapper.instance();
        const render = instance.renderConvertToPrivateRow();
        expect(render).toBeFalsy();
    });

    test('should not render convert to private button currentChannel is already private', async () => {
        const props = Object.assign({}, baseProps);
        props.currentChannel.type = General.PRIVATE_CHANNEL;
        const wrapper = shallow(
            <ChannelInfo
                {...props}
            />,
            {context: {intl: intlMock}},
        );

        const instance = wrapper.instance();
        const render = instance.renderConvertToPrivateRow();
        expect(render).toBeFalsy();
    });

    test('should not render convert to private button when currentChannel is a default channel', async () => {
        const props = Object.assign({}, baseProps);
        props.currentChannel.name = General.DEFAULT_CHANNEL;
        const wrapper = shallow(
            <ChannelInfo
                {...props}
            />,
            {context: {intl: intlMock}},
        );

        const instance = wrapper.instance();
        const render = instance.renderConvertToPrivateRow();
        expect(render).toBeFalsy();
    });
});
