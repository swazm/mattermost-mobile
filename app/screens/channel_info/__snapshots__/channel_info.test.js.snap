// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channel_info should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": undefined,
        "flex": 1,
      }
    }
  >
    <ChannelInfoHeader
      createAt={123}
      creator="Creator"
      displayName="Channel Name"
      hasGuests={false}
      header=""
      isArchived={false}
      isBot={false}
      isGroupConstrained={false}
      memberCount={2}
      onPermalinkPress={[Function]}
      purpose="Purpose"
      status="status"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
      type="O"
    />
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "borderTopColor": undefined,
          "borderTopWidth": 1,
        }
      }
    >
      <React.Fragment>
        <channelInfoRow
          action={[Function]}
          defaultMessage="Favorite"
          detail={false}
          icon="star-o"
          isLandscape={false}
          rightArrow={true}
          shouldRender={true}
          textId="mobile.routes.channelInfo.favorite"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          togglable={true}
        />
        <View
          style={
            Object {
              "backgroundColor": undefined,
              "height": 1,
              "marginHorizontal": 15,
            }
          }
        />
        <channelInfoRow
          action={[Function]}
          defaultMessage="Mute channel"
          detail={false}
          icon="bell-slash-o"
          isLandscape={false}
          rightArrow={true}
          shouldRender={true}
          textId="channel_notifications.muteChannel.settings"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          togglable={true}
        />
        <View
          style={
            Object {
              "backgroundColor": undefined,
              "height": 1,
              "marginHorizontal": 15,
            }
          }
        />
        <channelInfoRow
          action={[Function]}
          defaultMessage="Ignore @channel, @here, @all"
          detail={false}
          icon="at"
          isLandscape={false}
          rightArrow={true}
          shouldRender={true}
          textId="channel_notifications.ignoreChannelMentions.settings"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          togglable={true}
        />
        <View
          style={
            Object {
              "backgroundColor": undefined,
              "height": 1,
              "marginHorizontal": 15,
            }
          }
        />
        <channelInfoRow
          action={[Function]}
          defaultMessage="Pinned Posts"
          image={1}
          isLandscape={false}
          rightArrow={true}
          shouldRender={true}
          textId="channel_header.pinnedPosts"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          togglable={false}
        />
        <React.Fragment>
          <View
            style={
              Object {
                "backgroundColor": undefined,
                "height": 1,
                "marginHorizontal": 15,
              }
            }
          />
          <channelInfoRow
            action={[Function]}
            defaultMessage="Manage Members"
            detail={2}
            icon="users"
            isLandscape={false}
            rightArrow={true}
            shouldRender={true}
            textId="channel_header.manageMembers"
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
            togglable={false}
          />
        </React.Fragment>
        <React.Fragment>
          <View
            style={
              Object {
                "backgroundColor": undefined,
                "height": 1,
                "marginHorizontal": 15,
              }
            }
          />
          <channelInfoRow
            action={[Function]}
            defaultMessage="Add Members"
            icon="user-plus"
            isLandscape={false}
            rightArrow={true}
            shouldRender={true}
            textId="channel_header.addMembers"
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
            togglable={false}
          />
        </React.Fragment>
        <React.Fragment>
          <View
            style={
              Object {
                "backgroundColor": undefined,
                "height": 1,
                "marginHorizontal": 15,
              }
            }
          />
          <channelInfoRow
            action={[Function]}
            defaultMessage="Convert to Private Channel"
            icon="lock"
            isLandscape={false}
            rightArrow={true}
            shouldRender={true}
            textId="mobile.channel_info.convert"
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
            togglable={false}
          />
        </React.Fragment>
        <React.Fragment>
          <View
            style={
              Object {
                "backgroundColor": undefined,
                "height": 1,
                "marginHorizontal": 15,
              }
            }
          />
          <channelInfoRow
            action={[Function]}
            defaultMessage="Edit Channel"
            icon="edit"
            isLandscape={false}
            rightArrow={true}
            shouldRender={true}
            textId="mobile.channel_info.edit"
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
            togglable={false}
          />
        </React.Fragment>
      </React.Fragment>
      <React.Fragment>
        <View
          style={
            Object {
              "backgroundColor": undefined,
              "height": 1,
              "marginHorizontal": 15,
            }
          }
        />
        <channelInfoRow
          action={[Function]}
          defaultMessage="Leave Channel"
          icon="sign-out"
          isLandscape={false}
          rightArrow={true}
          shouldRender={true}
          textId="navbar.leave"
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          togglable={false}
        />
      </React.Fragment>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": undefined,
            "borderBottomWidth": 1,
            "borderTopColor": undefined,
            "borderTopWidth": 1,
          },
          Object {
            "borderBottomColor": undefined,
            "borderBottomWidth": 1,
            "borderTopColor": undefined,
            "borderTopWidth": 1,
            "marginTop": 40,
          },
        ]
      }
    >
      <channelInfoRow
        action={[Function]}
        defaultMessage="Archive Channel"
        icon="archive"
        iconColor="#CA3B27"
        isLandscape={false}
        rightArrow={true}
        shouldRender={true}
        textColor="#CA3B27"
        textId="mobile.routes.channelInfo.delete_channel"
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
        togglable={false}
      />
    </View>
  </ScrollViewMock>
</View>
`;
